public/js/components/Scopes.js

```diff
@@ -71,15 +73,99 @@ function getScopes(pauseInfo) {
       scopes.push([
         scope.getIn(["object", "class"]),
         fromJS({
+          isScope: true,
           value: scope.get("object")
         })
       ]);
     }
+    index++;
   } while (scope = scope.get("parent")); // eslint-disable-line no-cond-assign
 
   return scopes;
 }
 
+function isObject(obj) {
+  return obj.getIn(["value", "type"]) === "object";
+}
+
+function loadedProperties(obj, loadedObjects) {
+  return loadedObjects.get(obj.getIn(["value", "actor"])) ||
+         obj.get("prefetchedProperties");
+}
+
+function sortProperties(a, b) {
+  const obj = isObject(a[1]) && !isObject(b[1]) ? -10 : 10;
+  const alpha = a[0] < b[0] ? -1 : 1
+
+  return obj + alpha > 0 ? -1 : 1;
+}
+
+/**
+ * Gets an object's children to show in the scope tree
+ */
+function getChildren(item, loadedObjects) {
+  const obj = item[1];
+
+  if (isObject(obj)) {
+    const loaded = loadedProperties(obj, loadedObjects)
+
+    if (loaded) {
+      loaded.sort(sortProperties);
+      return loaded;
+    }
+
+    return [["fetching...", fromJS({ value: { type: "placeholder" }})]];
+  }
+
+  return [];
+}
+
+function runderItemValue(obj) {
+  let objectValue = "";
+  let hasValue = false;
+  if (obj.has("value")) {
+    hasValue = true;
+    let val = obj.get("value");
+    val = val.toJS ? val.toJS() : val;
+    if (isObject(obj)) {
+      objectValue = "Object";
+    } else {
+      objectValue = Rep({ object: val });
+    }
+  }
+  const isScope = obj.get("isScope")
+
+  if (isScope) {
+    return []
+  }
+
+  return [
+    dom.span({ className: "scope-object-delimiter" }, hasValue ? ": " : ""),
+    dom.span({ className: "scope-object-value" }, objectValue)
+  ]
+}
+
+function renderItem(item, depth, focused, _, expanded, { setExpanded }) {
+  const obj = item[1];
+
+  return dom.div(
+    { className: classnames("node", { focused }),
+      style: { marginLeft: depth * 15 }},
+    Arrow({
+      className: classnames(
+        { expanded: expanded,
+          hidden: item[1].getIn(["value", "type"]) !== "object" }
+      ),
+      onClick: e => {
+        e.stopPropagation();
+        setExpanded(item, !expanded);
+      }
+    }),
+    dom.span({ className: "scope-object-label" }, item[0]),
+    ...runderItemValue(obj)
+  );
+}
+
```

Comment:
I moved these functions out of the class as they didn't depend on props. I believe you have an idea for how you want to organize Scopes and I don't want to get in your way. Feel free to change this up.